src/agent/communicator/test/communicator_test.cpp

```diff
@@ -0,0 +1,17 @@
+#include "communicator.hpp"
+#include <gtest/gtest.h>
+
+using namespace communicator;
+
+TEST(CommunicatorTest, GetTokenRemainingSecsTest)
```

Comment:
Complete tests.

Reply:
A refactor is necessary to be able to do UT and components tests.